File: development/palindrome_word/palindrome_word.rb
def is_palindrome(word)
    left = 0
    right = word.length - 1

    while left < right
        return false if word[left] != word[right]

        left += 1
        right -= 1
    end

    true
end

word1 = "deified"
word2 = "neural"
puts "the word: #{word1} is a palindrome: #{is_palindrome(word1)}"
puts "the word: #{word2} is a palindrome: #{is_palindrome(word2)}"

File: development/palindrome_word/palindrome_word.js
function isPalindrome(word)
{
    let left = 0;
    let right = word.length - 1;

    while (left < right) {
        if (word[left] != word[right]) return false;

        left++;
        right--;
    }

    return true;
}

let word1 = "rotator"
let word2 =  "data"
console.log(`the word: ${word1} is palindrome: ${isPalindrome(word1)}`);
console.log(`the word: ${word2} is palindrome: ${isPalindrome(word2)}`);